hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/TestKeyManagerUnit.java

```diff
@@ -177,4 +194,128 @@ private OmMultipartInfo initMultipartUpload(KeyManagerImpl omtest,
+        new ScmClient(blockClient, containerClient),
+        metadataManager, configuration, "test-om",
+        Mockito.mock(OzoneBlockTokenSecretManager.class), null, null);
+
+    final DatanodeDetails dnOne = TestUtils.randomDatanodeDetails();
+    final DatanodeDetails dnTwo = TestUtils.randomDatanodeDetails();
+    final DatanodeDetails dnThree = TestUtils.randomDatanodeDetails();
+
+    final DatanodeDetails dnFour = TestUtils.randomDatanodeDetails();
+    final DatanodeDetails dnFive = TestUtils.randomDatanodeDetails();
+    final DatanodeDetails dnSix = TestUtils.randomDatanodeDetails();
+
+    final PipelineID pipelineOneId = PipelineID.randomId();
+    final Pipeline pipelineOne = Pipeline.newBuilder()
+        .setId(pipelineOneId)
+        .setType(ReplicationType.RATIS)
+        .setFactor(ReplicationFactor.THREE)
+        .setState(Pipeline.PipelineState.OPEN)
+        .setLeaderId(dnOne.getUuid())
+        .setNodes(Arrays.asList(dnOne, dnTwo, dnThree))
+        .build();
+
+    final PipelineID pipelineTwoId = PipelineID.randomId();
+    final Pipeline pipelineTwo = Pipeline.newBuilder()
+        .setId(pipelineOneId)
+        .setType(ReplicationType.RATIS)
+        .setFactor(ReplicationFactor.THREE)
+        .setState(Pipeline.PipelineState.OPEN)
+        .setLeaderId(dnFour.getUuid())
+        .setNodes(Arrays.asList(dnFour, dnFive, dnSix))
+        .build();
+
+    final ContainerBlockID blockID = new ContainerBlockID(1L, 1L);
+
+    final List<AllocatedBlock> blocks = new ArrayList<>();
+    final AllocatedBlock block = new AllocatedBlock.Builder()
+        .setPipeline(pipelineOne)
+        .setContainerBlockID(blockID)
+        .build();
+    blocks.add(block);
+
+    Mockito.when(blockClient.allocateBlock(Mockito.anyLong(), Mockito.anyInt(),
+        Mockito.any(ReplicationType.class),
+        Mockito.any(HddsProtos.ReplicationFactor.class),
+        Mockito.anyString(), Mockito.any(ExcludeList.class)))
+        .thenReturn(blocks);
+
+    Mockito.when(containerClient.getContainerWithPipeline(1L))
+        .thenReturn(new ContainerWithPipeline(null, pipelineTwo));
+
+    final OmVolumeArgs volumeArgs = OmVolumeArgs.newBuilder()
+        .setVolume("volumeOne")
+        .setAdminName("admin")
+        .setOwnerName("admin")
+        .build();
+    TestOMRequestUtils.addVolumeToOM(metadataManager, volumeArgs);
+
+    final OmBucketInfo bucketInfo = OmBucketInfo.newBuilder()
+          .setVolumeName("volumeOne")
+          .setBucketName("bucketOne")
+          .build();
+    TestOMRequestUtils.addBucketToOM(metadataManager, bucketInfo);
+
+    final OmKeyArgs.Builder keyArgs = new OmKeyArgs.Builder()
+        .setVolumeName("volumeOne")
+        .setBucketName("bucketOne")
+        .setKeyName("keyOne")
+        .setFactor(ReplicationFactor.THREE)
+        .setAcls(Collections.emptyList());
+
+    final OpenKeySession session = manager.createFile(keyArgs.build(),
+        false, false);
```

Comment:
Write requests in old code will be soon removed. Can we use similar API's which are used to create bucket/volume or directly use OmMetadataManager keyTable put to add key?

Reply:
Agreed, even the read path in old code will also be removed, this change is just to make sure that we are not breaking any functionality in the old code path. Both the KeyManagerImpl and the corresponding test class will be removed eventually.

Reply:
I think it will be better to add a similar API to create Key in TestOMRequestUtils, which will be helpful in future tests. (As right now there is no API to add the key to DB which takes OmKeyInfo as the argument.

Reply:
> even the read path in old code will also be removed

 We have not done any work for that, Read requests are using old code path. I am not sure there is any plan for rewrite read requests, as they can be used as-is for OM HA/non-HA

Reply:
`make sure that we are not breaking any functionality in the old code path`
For old code path for write requests which is no more used, I think it is okay to not to use these API's and use an API to create a key in DB by adding a new API in TestOmRequestUtils. As this is a newly added test, it is okay to do this way.